### PR TITLE
docs(header-navigation): adds yard to docs page

### PR DESCRIPTION
**Pull request by chasestarr**

### CODE REVIEW

File: documentation-site/components/yard/config/header-navigation.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+<StyledLink href="#basic-link2">Tab Link Two</StyledLink>
+</StyledNavigationItem>
+</StyledNavigationList>
+<StyledNavigationList $align={ALIGN.right}>
+<StyledNavigationItem>
+<Button>Get started</Button>
+</StyledNavigationItem>
+</StyledNavigationList>
+`,
+      type: PropTypes.String,
```
Comment (tajo) [inline documentation-site/components/yard/config/header-navigation.ts]:
```suggestion
      type: PropTypes.ReactNode,
```

File: documentation-site/components/yard/config/header-navigation.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  imports: {
+    'baseui/header-navigation': {
+      named: ['HeaderNavigation', 'ALIGN'],
+    },
+    'baseui/link': {
+      named: ['StyledLink'],
+    },
+    'baseui/button': {
+      named: ['Button'],
+    },
```
Comment (tajo) [inline documentation-site/components/yard/config/header-navigation.ts]: Could be moved to the children prop but doesn't really matter in this case.

File: documentation-site/components/yard/config/header-navigation.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+</StyledNavigationItem>
+<StyledNavigationItem>
+<StyledLink href="#basic-link2">Tab Link Two</StyledLink>
+</StyledNavigationItem>
+</StyledNavigationList>
+<StyledNavigationList $align={ALIGN.right}>
+<StyledNavigationItem>
+<Button>Get started</Button>
+</StyledNavigationItem>
+</StyledNavigationList>
```
Comment (tajo) [inline documentation-site/components/yard/config/header-navigation.ts]: (welp, we'll probably need to do something about the initial knob formatting through prettier)

Comment (chasestarr) [reply documentation-site/components/yard/config/header-navigation.ts]: maybe we could mark a prop to take up the full width of the container? (span both columns)

Comment (tajo) [reply documentation-site/components/yard/config/header-navigation.ts]: Yea, it's just there's not much value to edit the children knob vs edit the code snippet at that point.

Comment (chasestarr) [reply documentation-site/components/yard/config/header-navigation.ts]: good point

**Pull request merged**